pandas/formats/format.py

```diff
@@ -1951,7 +1951,11 @@ def _format_hierarchical_rows(self):
                 for spans, levels, labels in zip(level_lengths,
                                                  self.df.index.levels,
                                                  self.df.index.labels):
-                    values = levels.take(labels)
+
+                    values = levels.take(labels,
+                                         allow_fill=levels._can_hold_na,
```

Comment:
I didn't claim that very strongly :-) 

I only ran the specific test here in this PR, and there it is not catched, but as you show it is certainly catched in the other tests, so OK then!